parquet-cli/src/main/java/org/apache/parquet/cli/commands/ToAvroCommand.java

```diff
@@ -92,27 +92,22 @@ public int run() throws IOException {
     } else {
       schema = getAvroSchema(source);
     }
-    final Schema projection = filterSchema(schema, columns);
-
-    Path outPath = qualifiedPath(outputPath);
-    try (FileSystem outFS = outPath.getFileSystem(getConf())) {
-      if (overwrite && outFS.exists(outPath)) {
-        console.debug("Deleting output file {} (already exists)", outPath);
-        outFS.delete(outPath);
-      }
-    }
 
+    final Schema projection = filterSchema(schema, columns);
     Iterable<Record> reader = openDataFile(source, projection);
     boolean threw = true;
     long count = 0;
 
     DatumWriter<Record> datumWriter = new GenericDatumWriter<>(schema);
     try (DataFileWriter<Record> fileWriter = new DataFileWriter<>(datumWriter)) {
       fileWriter.setCodec(codecFactory);
-      try (DataFileWriter<Record> writer=fileWriter.create(projection, create(outputPath))) {
-        for (Record record : reader) {
-          writer.append(record);
-          count += 1;
+      try (OutputStream os = overwrite ?
```

Comment:
I think we can combine the two `try`'s here

Reply:
Sorry that I missed your comment @Fokko! Just updated the PR.